src/main/java/internet/shop/controller/product/AddProductController.java

```diff
@@ -0,0 +1,32 @@
+package internet.shop.controller.product;
+
+import internet.shop.lib.Injector;
+import internet.shop.model.Product;
+import internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) {
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
```

Comment:
What if we receive the empty Strings?

Reply:
We can't because we have a "required" parameter on the input tag in JSP)

Reply:
Actually, the user can do this by deleting the 'required block'
It's better to check everything that may cause internal server errors in the 'back-end code'
I'm not qualified enough, but I know that it's possible
For details, you can google this information :)